src/dag/dag_block_manager.cpp

```diff
@@ -0,0 +1,268 @@
+#include "dag_block_manager.hpp"
+
+namespace taraxa {
+
+DagBlockManager::DagBlockManager(addr_t node_addr, vdf_sortition::VdfConfig const &vdf_config,
+                                 optional<state_api::DPOSConfig> dpos_config, size_t capacity, unsigned num_verifiers,
+                                 std::shared_ptr<DbStorage> db, std::shared_ptr<TransactionManager> trx_mgr,
+                                 std::shared_ptr<FinalChain> final_chain, std::shared_ptr<PbftChain> pbft_chain,
+                                 logger::Logger log_time, uint32_t queue_limit)
+    : vdf_config_(vdf_config),
+      dpos_config_(dpos_config),
+      capacity_(capacity),
+      num_verifiers_(num_verifiers),
+      db_(db),
+      trx_mgr_(trx_mgr),
+      final_chain_(final_chain),
+      pbft_chain_(pbft_chain),
+      log_time_(log_time),
+      queue_limit_(queue_limit),
+      blk_status_(10000, 100),
+      seen_blocks_(10000, 100) {
+  LOG_OBJECTS_CREATE("BLKQU");
+}
+
+DagBlockManager::~DagBlockManager() { stop(); }
+
+void DagBlockManager::start() {
+  if (bool b = true; !stopped_.compare_exchange_strong(b, !b)) {
+    return;
+  }
+  LOG(log_nf_) << "Create verifier threads = " << num_verifiers_ << std::endl;
+  verifiers_.clear();
+  for (auto i = 0; i < num_verifiers_; ++i) {
+    verifiers_.emplace_back([this, i]() { this->verifyBlock(); });
+  }
+}
+
+void DagBlockManager::stop() {
+  if (bool b = false; !stopped_.compare_exchange_strong(b, !b)) {
+    return;
+  }
+  cond_for_unverified_qu_.notify_all();
+  cond_for_verified_qu_.notify_all();
+  for (auto &t : verifiers_) {
+    t.join();
+  }
+}
+
+bool DagBlockManager::isBlockKnown(blk_hash_t const &hash) {
+  auto known = seen_blocks_.count(hash);
+  if (!known) return getDagBlock(hash) != nullptr;
+  return true;
+}
+
+std::shared_ptr<DagBlock> DagBlockManager::getDagBlock(blk_hash_t const &hash) const {
+  std::shared_ptr<DagBlock> ret;
+  auto blk = seen_blocks_.get(hash);
+  if (blk.second) {
+    ret = std::make_shared<DagBlock>(blk.first);
+  }
+  if (!ret) {
+    return db_->getDagBlock(hash);
+  }
+  return ret;
```

Comment:
why not just write: 
if (blk.second) {
    return std::make_shared<DagBlock>(blk.first);
  }
   
  return db_->getDagBlock(hash);

Reply:
Right, fixed